### ISSUE TITLE
Void pointers in function parameters make building mock failing

### ISSUE THREAD

**Issue by Christer-Ekholm** [DrCpp/DrMock#24]
If I studied correctly then the following code have problems to be mocked.

class A
{
     void * b(void * a, const void *b);
}

ISEQUAL template cannot compare void poiters.

**Comment by maltekliemann** [comment]
That's true. The interface of `IsEqual` is not made to handle this situation.

Could you tell me how exactly you would compare void pointers? The only immediate option I see is to check if both point to the same address.

If you have a rough idea what type of data these pointers refer to, then I recommend using a wrapper struct which contains the data and can be handled by our framework.

**Comment by Christer-Ekholm** [comment]
I am working with embedded software and those void pointers are related to transfer buffers. Like UART_write(int fd, const void * data, int len);

I would do void pointer comparison so that unit test writer can decide if just compare if same address or maybe UT writer can give lambda function for testing equality of void pointers.

**Comment by maltekliemann** [comment]
Hmm, okay. I don't have the time to dive into this, but you may be able to solve this by writing a custom implementation of `IMatcher` (see https://github.com/DrCpp/DrMock/blob/master/docs/tutorial/mock.md#matching-and-polymorphism)
